src/System.Management.Automation/FormatAndOutput/common/ComplexWriter.cs

```diff
@@ -596,9 +596,9 @@ private static StringCollection GenerateLinesWithWordWrap(DisplayCells displayCe
                     string suffix = null;
 
                     // Handle soft hyphen
-                    if (word.Delim.Length == 1 && word.Delim[0] == s_softHyphen)
+                    if (word.Delim.Length == 1 && word.Delim[0] is SoftHyphen)
                     {
-                        int wordWidthWithHyphen = displayCells.Length(wordToAdd) + displayCells.Length(s_softHyphen);
+                        int wordWidthWithHyphen = displayCells.Length(wordToAdd) + displayCells.Length(SoftHyphen);
```

Comment:
`displayCells.Length(SoftHyphen)` - it seems to be a constant and we could exclude the call.

Reply:
`DisplayCells.Length(String, Int32)` is virtual method, I prefer to leave out this change to keep the PR simple.